ee/server/service/mdm.go

```diff
@@ -842,36 +830,90 @@ func (svc *Service) MDMAppleMatchPreassignment(ctx context.Context, externalHost
 		// TODO: seems like we don't support enabling disk encryption
 		// on team creation?
 		// see https://github.com/fleetdm/fleet/issues/12220
-		tm, err = svc.ModifyTeam(ctx, tm.ID, payload)
+		team, err = svc.ModifyTeam(ctx, team.ID, payload)
 		if err != nil {
 			return err
 		}
+	}
 
-		// create profiles for that team via the service call, so that uniqueness
-		// of profile identifier/name is verified, activity created, etc.
-		// NOTE: this will use the read replica to load the team, which was just
-		// created above, could lead to not found issues with slow replication.
-		if err := svc.BatchSetMDMAppleProfiles(ctx, &tm.ID, nil, rawProfiles, false); err != nil {
-			return err
-		}
-
-		targetTeamID = tm.ID
+	// create profiles for that team via the service call, so that uniqueness
+	// of profile identifier/name is verified, activity created, etc.
+	if err := svc.BatchSetMDMAppleProfiles(ctx, &team.ID, nil, rawProfiles, false); err != nil {
+		return err
 	}
 
 	// assign host to that team via the service call, which will trigger
 	// deployment of the profiles.
-	if err := svc.AddHostsToTeam(ctx, &targetTeamID, []uint{host.ID}); err != nil {
+	if err := svc.AddHostsToTeam(ctx, &team.ID, []uint{host.ID}); err != nil {
 		return err
 	}
+
 	return nil
+
+	//	// find a team with exactly that set of profiles
```

Comment:
Did you mean to leave this commented code?

Reply:
LOL, no, thank you! and I swear I did a "self review" and somehow didn't notice. I will remove.